Aur build fails (Arch Linux)

When attempting to build this error is displayed:

error[E0554]: `#![feature]` may not be used on the stable release channel
 --> src/main.rs:1:12
  |
1 | #![feature(backtrace_frames)]
  |            ^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0554`.
error: could not compile `drophost` due to previous error

--- a/.vscode/settings.json
+++ b/.vscode/settings.json
@@ -1,0 +1,3 @@
+{
+    "rust-analyzer.cargo.features": "all"
+}

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -24,12 +24,14 @@ tokio = { version = "1.24.2", features = ["macros", "rt", "rt-multi-thread"] }
 notify = "5.0.0"
 once_cell = "1.17.0"
 env_logger = "0.10.0"
+backtrace = { version = "0.3.62", optional = true, features = ["std"] }
 
 [features]
 default = []
 ping = ["dep:oping"]
 range = ["dep:iprange", "dep:network-interface", "dep:ipnet"]
 interface = ["dep:network-interface"]
+backtrace = ["dep:backtrace"]
 
 [dev-dependencies]
 compile_warning = { path = "compile_warning", version = "0.1.0" }

--- a/src/main.rs
+++ b/src/main.rs
@@ -1,4 +1,3 @@
-#![feature(backtrace_frames)]
 #![allow(dead_code)]
 #![allow(unused_imports)]
 
@@ -21,7 +20,7 @@ use nix::{unistd::Uid, sys};
 use compile_warning::compile_warning;
 #[cfg(not(dev))]
 #[allow(unused_macros)]
-macro_rules! compile_warning {() => {};}
+macro_rules! compile_warning {($($tt:tt)*) => {}}
 
 #[cfg(feature = "interface")]
 compile_warning!(The "interface" feature is not currently implemented.);

--- a/src/main.rs
+++ b/src/main.rs
@@ -5,6 +5,7 @@ mod cli;
 mod writer;
 mod types;
 mod parser;
+#[macro_use]
 mod utils;
 mod tests;
 

--- a/src/parser.rs
+++ b/src/parser.rs
@@ -5,7 +5,10 @@ use log::*;
 use compile_warning::compile_warning;
 #[cfg(not(dev))]
 #[allow(unused_macros)]
-macro_rules! compile_warning {() => {};}
+macro_rules! compile_warning {
+    // Take any number of arguments and ignore them
+    ($($arg:tt)*) => {};
+}
 
 use std::collections::HashMap;
 use std::fs::{self, File};

--- a/src/utils.rs
+++ b/src/utils.rs
@@ -1,7 +1,7 @@
 use std::fmt::Debug;
-use std::{error::Error, fmt::Display, backtrace};
+use std::{error::Error, fmt::Display};
 use log::*;
-use backtrace::BacktraceStatus::*;
+#[cfg(feature = "backtrace")]
 use backtrace::Backtrace;
 
 #[derive(Debug)]
@@ -31,30 +31,18 @@ impl Error for CustomError {
     }
 }
 
-
+#[cfg(feature = "backtrace")]
 macro_rules! print_trace {
     () => {
-        let backtrace = Backtrace::capture();
-        match backtrace.status() {
-        Captured => {
-            for line in backtrace.to_string().lines() {
-                trace!("At: {}", line);
-            }
-        },
-
-        Disabled => {
-            trace!("Backtrace disabled, set RUST_BACKTRACE=1 to enable");
-        }
-
-        Unsupported => {
-            trace!("Backtrace unsupported");
-            debug!("Running on platform: {}", std::env::consts::OS);
-        }
-
-        _ => {
-            warn!("An unknown error occurred while capturing the backtrace")
-        }
+        let backtrace = Backtrace::new();
+        trace!("Backtrace: {:?}", backtrace);
     }
+}
+
+#[cfg(not(feature = "backtrace"))]
+macro_rules! print_trace {
+    () => {
+        trace!("Backtrace disabled, enable with the 'backtrace' feature");
     };
 }
 

--- a/src/utils.rs
+++ b/src/utils.rs
@@ -1,8 +1,6 @@
 use std::fmt::Debug;
 use std::{error::Error, fmt::Display};
 use log::*;
-#[cfg(feature = "backtrace")]
-use backtrace::Backtrace;
 
 #[derive(Debug)]
 struct CustomError {
@@ -34,8 +32,16 @@ impl Error for CustomError {
 #[cfg(feature = "backtrace")]
 macro_rules! print_trace {
     () => {
+        use backtrace::Backtrace;
         let backtrace = Backtrace::new();
-        trace!("Backtrace: {:?}", backtrace);
+        
+        trace!("Backtrace:");
+        // Get the Debug representation of the backtrace
+        let debug = format!("{:?}", backtrace);
+        // Print each line individually
+        for line in debug.lines() {
+            trace!("{}", line);
+        }
     }
 }
 

